packages/engine/src/xr/XRControllerSystem.ts

```diff
@@ -105,7 +105,13 @@ export function updateGamepadInput(world: World, source: XRInputSource) {
       for (let i = 0; i < buttons.length; i++) {
         const buttonMapping = mapping[i]
         const button = buttons[i]
-        world.buttons[buttonMapping] = button
+        if (!world.buttons[buttonMapping] && button.pressed) {
```

Comment:
Having different shapes is problematic for optimization. All button instances should be created the same exact way.